### DISCUSSION TITLE
Best way to add dg-publish: true to hundreds of notes?

### DISCUSSION THREAD

**Discussion by simonalveteg** [oleeskild/obsidian-digital-garden#238] [Q&A]
I have a couple hundred notes that I want to publish, but going through them all and adding dg-publish to the front matter manually would be very tedious, is there a simpler way to go about this?

Every note i want to publish is already tagged, so the best solution for me would be if I could just give the plugin a list of tags and have it publish all notes that are tagged with one or more of those tags, if that makes sense?

Interested in hearing how others go about adding notes to be published!

**Comment by macbikegeek** [comment]
👍🏻 👀

**Reply by shinanqu** [reply]
You could add a shortcut to
Digital Garden: Publish Flag
And then you open the note, use the shortcut, open next note, etc.
It's not a perfect solution, but it's still very fast.

**Reply by simonalveteg** [reply]
That's a good suggestion, but I would have to do it for 1472 notes - which is a bit too time consuming even with shortcuts 😅

**Comment by simonalveteg** [comment]
I came up with a solution using Templater, Dataview and MetaEdit. The following template prompts the user to enter the name of the tag that they want dg-publish: true to be set for:
```javascript
<%*
const {createYamlProperty } = this.app.plugins.plugins["metaedit"].api;
let dv = DataviewAPI;
let tag = await tp.system.prompt("Enter tag (without #)")
var files = dv.pages("#"+tag).file.map(file => tp.file.find_tfile(file.path));
let count = 0;
for (let file of files) {
	await createYamlProperty('dg-publish', true, file);
	count++;
}
console.log("Parsed "+count+" files.");
-%>
```
There's probably a way to get the list of files without using dataview, but I just spent a few hours using dataviewjs so that's the easiest way I knew how to do it.

**Reply by simonalveteg** [reply]
Here's a better Template that only requires dataview, no need for metaedit (using a function that can be used to set any field of the frontmatter to any value)
```
<%*
let tag = await tp.system.prompt("Enter tag (without #)")
var files = DataviewAPI.pages("#"+tag).file.map(file => tp.file.find_tfile(file.path));
let count = 0;
for (let file of files) {
	await setFrontmatter(file, "dg-publish", true);
	count++;
}
console.log("Parsed "+count+" files.");

function setFrontmatter(file, fmfield, thing) {
	try {
		this.app.fileManager.processFrontMatter(file, fm => fm[fmfield] = thing);
	} catch (error) {
		console.log(`Error adding frontmatter ${thing} to ${fmfield}.`)
	}
}
-%>
```

**Comment by oleeskild** [comment]
That's a really cool solution @simonalveteg. I've never thought of extracting methods from another plugin like you did with metaedit. 

I also created a python script to achieve the same thing, so for anyone wanting a script solution that can be run outside Obsidian, without having to install other extensions, you're welcome to use this, which will add a yaml property on markdown files with the specified tag: 
https://gist.github.com/oleeskild/bd8daed6bcd1cab4cde9a0390447b55d

**Comment by BahneGork** [comment]
I wrote a powershell script for adding something to the front of a file, for like 1200 files.
Maybe someone can modify it for the need. Im a total newb with powershell.

If i remember correctly, my files had 0 tags so i read file content into a variable called Content. 
Then create another variable Newstring, which contains the "---" and then the tags i wanted title: filename, tags: notes, season: summer, toc: true, "---".
then the Newstring is inserted infront of the old Content.

Perhaps it can be modified to find existing "---" save that in a variable string: taglinestart, rest of file content saved in variable string Content and then modify Newstring to dg-publish: true and set the content to: $taglinesstart, $newstrnig, $content. ?

`
$files = Get-ChildItem -path C:\ *.md -recurse -force
foreach ($file in $files)
{
   	$content = Get-Content $file.FullName
	$newString = "---`rtitle: "+$file.BaseName+"`rtags: notes`rseason: summer`rtoc: true`r---`r `n# "+$file.BaseName
	Write-Output "================================ `r `n inserting" $newString " in" $file.FullName
	Set-Content $file.FullName -value $newString, $content
}
`
